.travis.yml

```diff
@@ -1,40 +1,125 @@
+conditions: v1
+
 language: python
+python:
+- "2.6"
+- "2.7"
+- "3.4"
+- "3.5"
+- "3.6"
+- "pypy2.7-5.10.0"
+- "pypy3.5-5.10.1"
 
 sudo: false
 
-cache: pip
+cache:
+  pip: true
+  directories:
+  - $HOME/.cache/pre-commit
+  - $HOME/.pre-commit
+  - $HOME/Library/Caches/Homebrew
 
-matrix:
-  include:
-    - python: 2.6
-      env: TOXENV=py26
-    - python: 2.7
-      env: TOXENV=py27
-    - python: 3.4
-      env: TOXENV=py34
-    - python: 3.5
-      env: TOXENV=py35
-    - python: 3.6
-      env: TOXENV=py36
-    - python: 3.7-dev
-      env: TOXENV=py37
-    - python: pypy
-      env: TOXENV=pypy
+install:
+- python -m pip.__main__ install cython tox
 
 # build libyaml
+- pushd /tmp
```

Comment:
Travis doesn't stop after a command fails. THat's why this code was setup this way. You might be able add `set -e` to the top. See https://github.com/travis-ci/travis-ci/issues/1066

Reply:
I've separated it because this way commands are easier to explore in log. Yes, we can enable and disable it in the end for this block if needed. We could also cache this thing, so that it would be built only if needed.

Reply:
@ingydotnet I've done this.